### DISCUSSION TITLE
Does next.js modify the css files before serving them or does it serve them statically?

### DISCUSSION THREAD

**Discussion by omgMath** [vercel/next.js#19028] [Help] [answered]
Does next.js modify the css files before serving them or does it serve them statically?

**Some context:**

We got reports of the wrong font appearing on our site. I was able to reproduce the issue on my phone, but strangely not on my desktop.

In the failing case the problem was obvious:
When I checked the styles in the debugger, i saw the following: `font-family: Graph Condensed,Impact,Arial #000,Arial,sans-serif;`:
![failing](https://user-images.githubusercontent.com/24620670/98716967-b1c7ff80-238c-11eb-8ff7-84c717ef4347.jpg)

Then again, in my Firefox everything looked fine: `font-family: Graph Condensed, Impact, Arial black, Arial, sans-serif;`, while the initial reporter had the same problem in Firefox as well:
![working](https://user-images.githubusercontent.com/24620670/98717033-d328eb80-238c-11eb-964a-504263c35a06.jpeg)

I also checked the css compiled/built from scss on the server. Everything looked fine, no "Arial #000" in there.

I'm now trying to figure out, what could have caused this... Any help or ideas are appreciated...

**Comment by jamesmosier** [comment, marked as answer]
Next uses postcss and cssnano-simple to optimize the CSS. From what I've seen most issues with CSS are upstream in one of those repositories, although I have not been able to find anything related to as you mentioned.

**Reply by omgMath** [reply]
Thanks for the hint.
Do you know if cssnano-simple is used/applied on each request?
Or am I checking the wrong file. Because `build/static/css/324db4ee0095426f1709.css` looks good, but what the browser arrives is (sometimes) wrong...

**Reply by jamesmosier** [reply]
> on each request

No, cssnano would only run at build time.

Do you happen to have a small example repository showing your issue?

**Reply by omgMath** [reply]
Although the my previous issue is slightly different, I think the cause is the same:
Here you go: https://github.com/omgMath/black-to-hex-issue

**Reply by jamesmosier** [reply]
<img width="349" alt="Screen Shot 2020-11-12 at 9 35 34 PM" src="https://user-images.githubusercontent.com/2854919/99021605-073c1200-252f-11eb-85b0-d01f1db37a67.png">

The CSS file built by Next is the same one that is served to the browser. I'd try clearing your build folder first to see what is actually built.

Aside from that, yes this looks like an upstream issue with either cssnano or postcss. I haven't found the issue yet though.

Wrapping it in quotes ("Arial Black") looks to not error out as a temporary fix.

**Reply by omgMath** [reply]
> The CSS file built by Next is the same one that is served to the browser. I'd try clearing your build folder first to see what is actually built.

Hm, okay. Then my observed behaviour is even weirder. Because in one browser arrived "Arial black", in the other one "Arial #000", but the same css was delivered (see screenshots - it is the same css file served)

> Wrapping it in quotes ("Arial Black") looks to not error out as a temporary fix.

I've noted that as well. But the "maybe-reproduction" differs from the original issue:

- My orignal scss had the quotes around "Arial Black"
- The built css had the quotes removed, but still stated "Arial Black"
- In some browsers it arrived as "Arial Black", in others as "Arial #000"

I think the error looks way to similar to be caused by different things. Therefore fixing one error might fix the other one as well. This is also why I asked if there is something modifying the css on the way from the server  to the client...

> Aside from that, yes this looks like an upstream issue with either cssnano or postcss. I haven't found the issue yet though.

I've quickly threw my css against the web-cssnano, looked good so far. But not sure if I used the correct configuration...
